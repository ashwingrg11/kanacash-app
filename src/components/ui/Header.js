import React from 'react';
import {useNavigation, DrawerActions} from '@react-navigation/native';
import {
  View,
  StyleSheet,
  TouchableOpacity,
  Platform,
  Image,
} from 'react-native';
import theme from '~/components/theme/Style';
import {HamburgerSvgIcon} from '~/components/ui/Icon';
import {SemiBoldText} from '~/components/ui/Text';
import {BackButton} from '~/components/ui/Button';

export default function Header({title = '', backButtonVisible = false}) {
  const navigation = useNavigation();
  return (
    <View style={styles.header}>
      <View
        style={
          Platform.OS === 'ios'
            ? styles.headerContentIos
            : styles.headerContentAndroid
        }>
        <View style={styles.buttonNavigation}>
          {backButtonVisible ? (
            <TouchableOpacity style={styles.leftIcon}>
              <BackButton navigation={navigation} />
            </TouchableOpacity>
          ) : (
            <TouchableOpacity
              onPress={() => navigation.dispatch(DrawerActions.toggleDrawer())}
              style={styles.leftIcon}>
              <HamburgerSvgIcon />
            </TouchableOpacity>
          )}
        </View>
        <View style={styles.headerTitle}>
          <SemiBoldText text={title} style={styles.headerTitleText} />
        </View>
        <View style={styles.headerLogo}>
          <Image source={require('../../assets/image/HeaderLogo.png')} />
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    paddingTop: 10,
    alignItems: 'center',
  },
  headerContentIos: {
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  headerContentAndroid: {
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-between',
  },
  buttonNavigation: {},
  leftIcon: {
    width: 40,
    padding: 5,
    left: 14,
    bottom: 1,
  },
  headerLogo: {
    justifyContent: 'center',
    right: 20,
    bottom: 3,
  },
  headerTitle: {
    flex: 2,
    alignItems: 'center',
    justifyContent: 'center',
  },
  headerTitleText: {
    fontSize: 18,
    color: theme.secondaryColor,
  },
  backButtonStyle: {
    bottom: 18,
  },
});
