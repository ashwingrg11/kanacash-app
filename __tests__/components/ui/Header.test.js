import 'react-native';
import React from 'react';

import Header from '~/components/ui/Header';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<Header/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
