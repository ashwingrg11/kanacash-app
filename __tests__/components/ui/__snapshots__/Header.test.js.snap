// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 1,
            "left": 14,
            "opacity": 1,
            "padding": 10,
            "width": 40,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="12"
          bbWidth="18"
          fill="none"
          focusable={false}
          height="12"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 12,
                "width": 18,
              },
            ]
          }
          vbHeight={12}
          vbWidth={18}
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M17.05 5H0.95C0.42533 5 0 5.42533 0 5.95V6.05C0 6.57467 0.42533 7 0.95 7H17.05C17.5747 7 18 6.57467 18 6.05V5.95C18 5.42533 17.5747 5 17.05 5Z"
              fill={4280229663}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M17.05 10H0.95C0.42533 10 0 10.4253 0 10.95V11.05C0 11.5747 0.42533 12 0.95 12H17.05C17.5747 12 18 11.5747 18 11.05V10.95C18 10.4253 17.5747 10 17.05 10Z"
              fill={4280229663}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M17.05 0H0.95C0.42533 0 0 0.425329 0 0.95V1.05C0 1.57467 0.42533 2 0.95 2H17.05C17.5747 2 18 1.57467 18 1.05V0.95C18 0.425329 17.5747 0 17.05 0Z"
              fill={4280229663}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "OpenSans-SemiBold",
              "fontSize": 20,
              "lineHeight": 25,
            },
            undefined,
            Object {
              "color": "#0E3C65",
              "fontSize": 18,
            },
          ]
        }
        text=""
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "bottom": 3,
          "justifyContent": "center",
          "right": 20,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="19"
        bbWidth="40"
        fill="none"
        focusable={false}
        height="19"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 19,
              "width": 40,
            },
          ]
        }
        vbHeight={19}
        vbWidth={40}
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M34.3536 2.25232C34.7641 2.08411 35.2132 1.8727 35.6499 1.64509C36.2562 1.34482 36.8629 0.980536 37.4879 0.605271C37.7783 0.430889 38.0727 0.254136 38.3728 0.0803337L38.453 0.231562C37.6552 1.08492 36.8242 1.88582 35.9523 2.58564C35.0781 3.27466 34.1646 3.85874 33.2295 4.27539C32.2943 4.69204 31.3422 4.94203 30.4117 5.02845C29.4788 5.11795 28.5668 5.05005 27.6865 4.87953C25.9249 4.5138 24.2985 3.75766 22.6967 2.98609C22.5083 2.89523 22.3201 2.80391 22.132 2.71261C21.5197 2.41545 20.9079 2.11852 20.2894 1.83875C19.4854 1.46454 18.6706 1.12196 17.8257 0.858082L17.889 0.655157C18.0743 0.625253 18.2564 0.614955 18.4392 0.604623C18.497 0.60136 18.5548 0.598092 18.6127 0.594203C18.8496 0.593431 19.0834 0.598832 19.3195 0.606548C19.7917 0.622751 20.2485 0.68602 20.7029 0.75932C21.6088 0.910549 22.4799 1.14819 23.3286 1.39818C23.7912 1.53848 24.2475 1.68406 24.699 1.82811C25.9013 2.21168 27.0693 2.58433 28.2304 2.81711C29.8283 3.14272 31.4132 3.18438 33.079 2.70369C33.1231 2.69014 33.1672 2.67673 33.2115 2.6633C33.3764 2.61324 33.5424 2.56286 33.7078 2.50385C33.9059 2.43424 34.0953 2.35728 34.3058 2.27174L34.3536 2.25232ZM35.8505 5.67734C34.8497 6.21898 33.866 6.66418 32.9077 6.95198C30.9911 7.5322 29.1393 7.516 27.313 7.1333C25.9244 6.84573 24.5411 6.36745 23.1186 5.87562C22.667 5.7195 22.2116 5.56201 21.7507 5.40883C20.7924 5.09557 19.8117 4.79697 18.7925 4.59482C18.2825 4.4976 17.7678 4.41041 17.2393 4.37646C16.9739 4.35871 16.7077 4.3456 16.4423 4.34097C16.3892 4.34264 16.3361 4.34398 16.283 4.34532C16.066 4.3508 15.8488 4.35628 15.6306 4.38418L15.5681 4.58864C16.5333 4.86795 17.4638 5.25143 18.3805 5.67656C19.1675 6.02975 19.9444 6.41525 20.7217 6.80099L20.7218 6.80104C20.8513 6.86528 20.9808 6.92953 21.1103 6.99364C22.9266 7.88713 24.766 8.77135 26.7629 9.19109C27.7621 9.39093 28.799 9.46886 29.8561 9.3701C30.9155 9.26748 31.9949 8.99434 33.0574 8.52368C34.1183 8.05302 35.1576 7.39563 36.1506 6.61789C37.1429 5.82934 38.0904 4.92814 38.9977 3.95595L38.9183 3.80395C38.66 3.96637 38.405 4.12975 38.1522 4.29164L38.1522 4.29168C37.3712 4.79197 36.6124 5.27803 35.8505 5.67734ZM8.66477 7.1527C9.47801 6.96135 10.2882 6.85796 11.0914 6.81475C11.2634 6.8055 11.434 6.80164 11.6029 6.80164C12.2287 6.80164 12.8436 6.85796 13.4539 6.93126C13.8413 6.98296 14.2263 7.04546 14.6059 7.12184C14.9863 7.20131 15.359 7.30239 15.7301 7.41041C16.4739 7.62028 17.1922 7.9019 17.8974 8.20668L19.9113 9.12562C20.574 9.4358 21.2368 9.73903 21.9019 10.0214C24.5523 11.1718 27.1926 12.1872 29.9888 12.0005C30.3391 11.9774 30.6894 11.9311 31.0443 11.8632C31.3808 11.7966 31.7179 11.7072 32.0571 11.6172L32.0571 11.6172L32.0572 11.6172L32.1129 11.6024C32.8321 11.3995 33.5566 11.097 34.2842 10.7251C35.7502 10.0075 37.2023 8.95819 38.6798 7.85484L38.7608 8.00684C37.4283 9.42191 36.0187 10.7097 34.5179 11.7119C33.0211 12.7096 31.437 13.3832 29.8792 13.6424C28.3168 13.9102 26.796 13.7852 25.3455 13.4572C23.8957 13.1262 22.5037 12.6031 21.1442 12.0159C19.7886 11.418 18.4592 10.749 17.1367 10.0785C15.8142 9.42037 14.4369 8.89801 13.0018 8.57086C11.5682 8.24526 10.0768 8.1164 8.5529 8.27766C7.79135 8.3525 7.02209 8.50373 6.2536 8.74755C5.48974 8.99754 4.712 9.32392 3.99057 9.82853L3.93656 9.65338C4.64641 8.94353 5.4365 8.4559 6.23509 8.04542C7.03367 7.63957 7.84922 7.3672 8.66477 7.1527ZM5.41337 11.1524C6.34389 10.9511 7.2667 10.8438 8.18024 10.8091C8.32607 10.8037 8.4719 10.8006 8.61618 10.8006C9.38004 10.8006 10.13 10.8762 10.873 10.9757C11.3136 11.0367 11.7526 11.1146 12.1855 11.2057C12.616 11.3021 13.0419 11.4209 13.464 11.5459C14.3104 11.7936 15.129 12.12 15.9346 12.4726L18.2462 13.5389C19.0085 13.8985 19.7716 14.2465 20.5378 14.5744C23.5886 15.8961 26.6556 17.0619 29.914 16.7996C30.3206 16.7649 30.7311 16.7055 31.1423 16.6206C31.5543 16.5349 31.9694 16.4208 32.3846 16.2988C33.2194 16.0496 34.0581 15.6854 34.8945 15.2402C36.5835 14.3792 38.2462 13.1408 39.9213 11.8245L40 11.9796C38.4723 13.6006 36.8504 15.0805 35.1298 16.2271C33.4107 17.3721 31.5945 18.1437 29.8036 18.4407C28.012 18.7471 26.2675 18.6012 24.6024 18.2247C22.9366 17.8443 21.3425 17.2433 19.7793 16.5681C18.2207 15.8822 16.6953 15.1137 15.1753 14.3444C13.6553 13.5883 12.0682 12.988 10.4209 12.6138C8.77281 12.2404 7.0576 12.093 5.30381 12.2774C4.42807 12.3677 3.5423 12.5405 2.65962 12.8237C1.7808 13.113 0.88654 13.488 0.056325 14.0705L0 13.8954C0.823271 13.1061 1.72756 12.5714 2.64033 12.1231C3.55465 11.6771 4.48517 11.3847 5.41337 11.1524Z"
            fill={4289929009}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M16.7035 0C16.7035 0 15.1264 1.66274 14.4405 3.75371C14.3564 4.00833 12.4421 2.81856 8.43225 4.87558C8.43225 4.87558 12.3171 0.810154 13.0146 0.476833C13.7114 0.144285 16.7035 0 16.7035 0Z"
            fill={4280229663}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
